ledger/voters.go

```diff
@@ -291,7 +291,30 @@ func (vt *votersTracker) lowestRound(base basics.Round) basics.Round {
 	return minRound
 }
 
-// VotersForStateProof returns the top online participants from round r.
+// LatestCompletedVotersUpTo returns the highest round <= r for which information about the top online
+// participants has already been collected,  and the completed VotersForRound for that round.
+// If none is found, it returns 0, nil. Unlike VotersForStateProof, this function does not wait.
+func (vt *votersTracker) LatestCompletedVotersUpTo(r basics.Round) (basics.Round, *ledgercore.VotersForRound) {
+	vt.votersMu.RLock()
+	defer vt.votersMu.RUnlock()
+
+	var latestRound basics.Round
+	var latestVoters *ledgercore.VotersForRound
+
+	for round, voters := range vt.votersForRoundCache {
+		if round <= r && round > latestRound {
+			if completed, err := voters.Completed(); completed && err == nil {
```

Comment:
Probably my lack of familiarity, but what causes rounds with errors present?

Reply:
Calling @cce, who wrote this.  But I think the idea is that an error is returned for the simple problem of the information not being collected yet.